src/main/java/com/tistory/povia/realworld/user/domain/User.java

```diff
@@ -0,0 +1,177 @@
+package com.tistory.povia.realworld.user.domain;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.tistory.povia.realworld.common.domain.BaseTimeEntity;
+import java.util.Objects;
+import javax.persistence.*;
+import org.apache.commons.lang3.StringUtils;
+import org.hibernate.annotations.Where;
+
+@Entity(name = "user")
+@Where(clause = "deleted_at is null")
+public class User extends BaseTimeEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+    @Embedded private Email email;
+
+    @Column(name = "username", length = 50, nullable = false)
+    private String username;
+
+    @Column(name = "password", nullable = false)
+    private String password;
+
+    @Column(name = "bio")
+    private String bio;
+
+    @Column(name = "image")
+    private String image;
+
+    protected User() {}
+
+    private User(Email email, String username, String password, String bio, String image) {
+        checkArgument(StringUtils.isNotBlank(password), "password should be provided");
+        checkUsername(username);
+        checkPassword(password);
```

Comment:
```suggestion
        checkUsername(username);
        checkPassword(password);
```
`checkPassword()` validate를 진행할 때 password에 대한 `isNotBlank`를 진행하는데 중복되는 로직이 아닐까요?

Reply:
중복되는 로직이 맞읍니다. 저도 지금 보고 삭제했네요